lua/telescope/builtin/files.lua

```diff
@@ -78,7 +78,16 @@ files.live_grep = function(opts)
         search_list = filelist
       end
 
-      return flatten { vimgrep_arguments, prompt, search_list }
+      local extraArgs = {}
+      local smart_mask = utils.get_default(opts.smart_mask, conf.smart_mask)
+      local filetype = opts.current_filetype or ''
+
+      if (smart_mask == true and filetype ~= '') then
```

Comment:
Only one option, so if we have a smart_mask table then its enabled, no table or table empty then disabled.

Reply:
Ah, but what happens when you _don't_ want smart mask to be applied? Example: In general, I want it to be on most of the time, but I can see some instances where I actually want it off right now since I do want to search all files. My thought process was passing the smart_mask as a param would allow me to map two bindings, one with mask on, one with off. 

Thoughts?